blockly/src/starter/PathfinderStarter.java

```diff
@@ -108,4 +106,55 @@ public static void createHero() throws IOException {
     hero.remove(CameraComponent.class);
     Game.add(hero);
   }
+
+  /**
+   * Checks if any key for selecting a pathfinding algorithm has been pressed.
+   *
+   * <p>This method listens for specific key presses for DFS, BFS, and other pathfinding algorithms.
+   * It switches the current algorithm accordingly and prevents further input until a new algorithm
+   * is selected. If the "SuS Pathfinding" algorithm is selected but not implemented, an {@link
+   * OperationNotSupportedException} is thrown to indicate the feature is not available.
+   */
+  private static void checkForAlgorithmSwitch() {
+    try {
+      Coordinate startCoords = Game.currentLevel().startTile().coordinate();
+      Coordinate endTileCoords = Game.currentLevel().endTile().coordinate();
+      if (noPathfindingSelected && Gdx.input.isKeyJustPressed(KeyboardConfig.SELECT_DFS.value())) {
+        switchToAlgorithm(new DFSPathFinding(startCoords, endTileCoords), "DFS-Pathfinding");
+      } else if (noPathfindingSelected
+          && Gdx.input.isKeyJustPressed(KeyboardConfig.SELECT_BFS.value())) {
+        switchToAlgorithm(new BFSPathFinding(startCoords, endTileCoords), "BFS-Pathfinding");
+      } else if (noPathfindingSelected
+          && Gdx.input.isKeyJustPressed(KeyboardConfig.SELECT_SUS_ALGO.value())) {
+        if (noSuSAlgorithmAvailable) {
+          throw new OperationNotSupportedException(
+              "The SuS Pathfinding algorithm is not implemented yet.");
+        }
+        // switchToAlgorithm(new SuSPathFinding(startCoords, endTileCoords), "SuS-Pathfinding");
+      }
+    } catch (OperationNotSupportedException e) {
+      System.err.println(e.getMessage());
+    }
+  }
+
+  /**
+   * Changes the pathfinding algorithm used in the game to the specified one.
+   *
+   * <p>This method updates the active pathfinding algorithm in the {@link PathfindingSystem} and
+   * updates the game window title to reflect the newly selected algorithm.
+   *
+   * @param algorithm The new pathfinding algorithm to be used (e.g., DFS, BFS, SuS).
+   * @param name The name of the algorithm to display in the window title for the player's
+   *     reference.
+   */
+  private static void switchToAlgorithm(utils.pathfinding.PathfindingLogic algorithm, String name) {
+    Game.system(
+        PathfindingSystem.class,
+        pfs -> {
+          pfs.autoStep(true);
+          pfs.updatePathfindingAlgorithm(algorithm);
+        });
+    Gdx.graphics.setTitle("Blockly KI-Dungeon – " + name);
```

Comment:
```suggestion
    Game.windowTitle("Blockly KI-Dungeon – " + name);
```
ist glaub ich sinnvoller, bevor wir mehr Abhängigkeiten von gdx hinzufügen

Reply:
Das wollte ich eigentlich auch verwenden, aber das funktioniert leider nicht. Der Titel aktualisiert sich damit nicht.

Reply:
> Das wollte ich eigentlich auch verwenden, aber das funktioniert leider nicht. Der Titel aktualisiert sich damit nicht.

Hmm ja, das setzt das nur für die PreRunConfig.

Aber erstell mal in Game eine funktion, die genau das macht.
Also 

```
public static void updateWindowTitle(String newTitle)
  Gdx.graphics.setTitle(newTitle);
 ```